R/enrichment_map.R

```diff
@@ -122,6 +124,25 @@ enrichment_map <- function(res_enrich,
   omm <- omm[omm$gs_1 != omm$gs_2, ]
   # ... and the ones from the other triangular portion
   omm <- omm[!is.na(omm$value), ]
+  
+  add_most_senior_node <- function(res_enrich, omm) {
+    onts <- unique(AnnotationDbi::select(GO.db::GO.db, 
+                                         keys = res_enrich$gs_id, 
+                                         "ONTOLOGY")$ONTOLOGY)
+    if(onts == "BP") {
+      go_senior <- "biological processes"
+    }  else if(onts == "MF") {
+      go_senior <- "molecular functions"
+    } else if(onts == "CC") {
+      go_senior <- "cellular components"
+    }
+    all_terms <- unique(c(omm$gs_1, omm$gs_2))
+    tmp_df <- data.frame("gs_1" = rep(go_senior, length(all_terms)), 
+                         "gs_2" = all_terms,
+                         "value" = rep(1, length(all_terms)))
+    omm <- rbind(omm, tmp_df)
+  }
```

Comment:
Would it make sense to have this function outside the function itself?
Or alternatively: this one does not seem to allow more than one ontology class

Another thing: this is restricted to GO, but the em can actually contain any geneset. Maybe it is worth making this optional/sensing whether the terms are GO?

Reply:
Plus, running the example on it does not seem to work - handling of NA is probably the culprit

Reply:
I think we need cbind more than rbind